src/main/webapp/recipe-display.html

```diff
@@ -7,44 +7,129 @@
     <!-- Markdown rendering -->
     <script src="https://cdn.jsdelivr.net/npm/marked/marked.min.js"></script>
   </head>
-  <body onload="getDetailedRecipe(getRecipeID()); getComments(getRecipeID())">
+  <body>
     <div id="recipe"></div>
     <!-- Comment submission form -->
     <div>
-      <label for="comment-content">Message</label><br />
+      <label for="comment-content" id="comment-label">Message</label><br />
       <input type="text" id="comment-content" name="comment-content" placeholder="Leave a comment..." />
       <br />
-      <button type="button" onclick="submitComment(getRecipeID())" value="Submit">Submit</button>
+      <button type="button" id="submit-comment" value="Submit">Submit</button>
     </div>
     <div id="comments"></div>
-    <button type="button" onclick="deletePost(getRecipeID())">Delete Post</button>
+    <button type="button" id="delete-post">Delete Post</button>
 
+    <!-- Firebase App (the core Firebase SDK) is always required and must be listed first -->
+    <script src="https://www.gstatic.com/firebasejs/7.16.1/firebase-app.js"></script>
+
+    <!-- If you enabled Analytics in your project, add the Firebase SDK for Analytics -->
+    <script src="https://www.gstatic.com/firebasejs/7.16.1/firebase-analytics.js"></script>
+
+    <!-- Add Firebase products that you want to use -->
+    <script src="https://www.gstatic.com/firebasejs/7.16.1/firebase-auth.js"></script>
+    
     <!-- Page scripts -->
-    <script>
+    <script type="module">
+      import { app } from "/scripts/init.js"
+
+      var deletePostButton = document.getElementById("delete-post")
+      var submitCommButton = document.getElementById("submit-comment")
+      var commentLabel = document.getElementById("comment-label")
+      var commentContent = document.getElementById("comment-content")
+
+      /** On window load, initialize the page. */
+      window.onload = function () {
+        initPage()
+      }
+
+      /** Load page elements. */
+      function initPage() {
+        
+        // Load the recipe and comments on the page.
+        getDetailedRecipe(getRecipeID())
+        getComments(getRecipeID())
+
+        // Set button click listeners.
+        deletePostButton.addEventListener("click", deletePost)
+        submitCommButton.addEventListener("click", submitComment)
+
+        // Hide buttons and comment input for non signed-in user.
+        toggleElementVisibility()
+
+        // Set auth state change listener.
+        // Currently toggles visibility only.
+        app.auth().onAuthStateChanged(function (user){
+          if (user) {
+            deletePostButton.hidden = false
+            submitCommButton.hidden = false
+            commentLabel.hidden = false
+            commentContent.hidden = false
+          }
+          else {
+            deletePostButton.hidden = true
+            submitCommButton.hidden = true
+            commentLabel.hidden = true
+            commentContent.hidden = true
```

Comment:
consider making a function updateButtons(isLoggedIn) or similar, which could potentially be like:
function updateButtons(/* bool */ isLoggedIn) {
  var els = [dPB, sCB, cL, CC];
  for (el in els) {
    el.hidden = !isLoggedIn'
  }
}
or similar (this is pseudocode :) )